scalyr_agent/test_base.py

```diff
@@ -322,3 +248,142 @@ def assertGreater(self, a, b, msg=None):
 
         def assertLess(self, a, b, msg=None):
             unittest.TestCase.assertLess(self, a, b, msg=msg)
+
+
+class BaseScalyrLogCaptureTestCase(ScalyrTestCase):
+    """
+    Base test class which captures log data produced by code called inside the tests into the log
+    files created in "directory_path" directory as defined by that class variable.
+
+    Directory available via "directory_path" variable is automatically created in a secure random
+    fashion for each test invocation inside setUp() method.
+
+    In addition to creating this directory, this method also sets up agent logging so it logs to
+    files in that directory.
+
+    On tearDown() the log directory is removed if tests pass and assertion hasn't failed, but if the
+    assertion fails, directory is left in place so developer can inspect the log content which might
+    aid with test troubleshooting.
+    """
+    # Path to the directory with the agent logs
+    logs_directory = None  # type: Optional[str]
+
+    # Path to the main agent log file
+    agent_log_path = None  # type: Optional[str]
+
+    # Path to the agent debug log file (populated inside setUp()
+    agent_debug_log_path = None  # type: Optional[str]
+
+    # Variable which indicates if assertLogFileContainsLine assertion was used and it fails
+    # NOTE: Due to us needing to support multiple Python versions and test runners, there is no easy
+    # and simple test runner agnostic way of detecting if tests have failed
+    __assertion_failed = False  # type: bool
+
+    def setUp(self):
+        super(BaseScalyrLogCaptureTestCase, self).setUp()
+
+        self.logs_directory = tempfile.mkdtemp(suffix='agent-tests-log')
+
+        scalyr_logging.set_log_destination(
+            use_disk=True,
+            logs_directory=self.logs_directory,
+            agent_log_file_path='agent.log',
+            agent_debug_log_file_suffix='_debug'
+        )
+        scalyr_logging.__log_manager__.set_log_level(scalyr_logging.DEBUG_LEVEL_5)
+
+        self.agent_log_path = os.path.join(self.logs_directory, 'agent.log')
+        self.agent_debug_log_path = os.path.join(self.logs_directory, 'agent_debug.log')
+
+    def tearDown(self):
+        super(BaseScalyrLogCaptureTestCase, self).tearDown()
+
+        if self.__assertion_failed:
+            # Print the paths to which we store the output to so they can be introspected by the
+            # developer
+            test_name = self._testMethodName
+            print('Storing agent log file for test "%s" to: %s' % (test_name, self.agent_log_path))
```

Comment:
Ok, maybe minor nit.. but shouldn't this be in the past tense now?  `Stored agent log file for test...`?  Same below.